compiler/parser/chapel.lex

```diff
@@ -154,7 +154,9 @@ param            return processToken(yyscanner, TPARAM);
 zip              return processToken(yyscanner, TZIP);
 pragma           return processToken(yyscanner, TPRAGMA);
 __primitive      return processToken(yyscanner, TPRIMITIVE);
+private          return processToken(yyscanner, TPRIVATE);
 proc             return processToken(yyscanner, TPROC);
+public           return processToken(yyscanner, TPUBLIC);
 record           return processToken(yyscanner, TRECORD);
 reduce           return processToken(yyscanner, TREDUCE);
```

Comment:
It just struck me that you have introduced public and private as keywords.  It's clear to me that this is
convenient for parsing, and it may well be appropriate.  However I believe there was some effort to
limit the number of keywords.  Just want to be sure (a) that this is a conscious choice (b) propose that
the spec be updated in the near future if these are to be keywords.

Also I know you didn't do this but I note that "zip" is the only keyword that's not in alphabetical order.
Would you be willing to fix that.  There's no need to re-test if you do make this change so long as the
parser still builds etc.